service/dynamodb/dynamodbattribute/empty_collections_test.go

```diff
@@ -0,0 +1,663 @@
+package dynamodbattribute
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+type testEmptyCollectionsNumericalScalars struct {
+	String string
+
+	Uint8  uint8
+	Uint16 uint16
+	Uint32 uint32
+	Uint64 uint64
+
+	Int8  int8
+	Int16 int16
+	Int32 int32
+	Int64 int64
+
+	Float32 float32
+	Float64 float64
+}
+
+type testEmptyCollectionsOmittedNumericalScalars struct {
+	String string `dynamodbav:",omitempty"`
+
+	Uint8  uint8  `dynamodbav:",omitempty"`
+	Uint16 uint16 `dynamodbav:",omitempty"`
+	Uint32 uint32 `dynamodbav:",omitempty"`
+	Uint64 uint64 `dynamodbav:",omitempty"`
+
+	Int8  int8  `dynamodbav:",omitempty"`
+	Int16 int16 `dynamodbav:",omitempty"`
+	Int32 int32 `dynamodbav:",omitempty"`
+	Int64 int64 `dynamodbav:",omitempty"`
+
+	Float32 float32 `dynamodbav:",omitempty"`
+	Float64 float64 `dynamodbav:",omitempty"`
+}
+
+type testEmptyCollectionsPtrScalars struct {
+	PtrString *string
+
+	PtrUint8  *uint8
+	PtrUint16 *uint16
+	PtrUint32 *uint32
+	PtrUint64 *uint64
+
+	PtrInt8  *int8
+	PtrInt16 *int16
+	PtrInt32 *int32
+	PtrInt64 *int64
+
+	PtrFloat32 *float32
+	PtrFloat64 *float64
+}
+
+type testEmptyCollectionsOmittedPtrNumericalScalars struct {
+	PtrUint8  *uint8  `dynamodbav:",omitempty"`
+	PtrUint16 *uint16 `dynamodbav:",omitempty"`
+	PtrUint32 *uint32 `dynamodbav:",omitempty"`
+	PtrUint64 *uint64 `dynamodbav:",omitempty"`
+
+	PtrInt8  *int8  `dynamodbav:",omitempty"`
+	PtrInt16 *int16 `dynamodbav:",omitempty"`
+	PtrInt32 *int32 `dynamodbav:",omitempty"`
+	PtrInt64 *int64 `dynamodbav:",omitempty"`
+
+	PtrFloat32 *float32 `dynamodbav:",omitempty"`
+	PtrFloat64 *float64 `dynamodbav:",omitempty"`
+}
+
+type testEmptyCollectionTypes struct {
+	Map       map[string]string
+	Slice     []string
+	ByteSlice []byte
+	ByteArray [4]byte
+	ZeroArray [0]byte
+	BinarySet [][]byte `dynamodbav:",binaryset"`
+	NumberSet []int    `dynamodbav:",numberset"`
+	StringSet []string `dynamodbav:",stringset"`
+}
+
+type testEmptyCollectionTypesOmitted struct {
+	Map       map[string]string `dynamodbav:",omitempty"`
+	Slice     []string          `dynamodbav:",omitempty"`
+	ByteSlice []byte            `dynamodbav:",omitempty"`
+	ByteArray [4]byte           `dynamodbav:",omitempty"`
+	ZeroArray [0]byte           `dynamodbav:",omitempty"`
+	BinarySet [][]byte          `dynamodbav:",binaryset,omitempty"`
+	NumberSet []int             `dynamodbav:",numberset,omitempty"`
+	StringSet []string          `dynamodbav:",stringset,omitempty"`
+}
+
+type testEmptyCollectionStruct struct {
+	Int int
+}
+
+type testEmptyCollectionStructOmitted struct {
+	Slice []string `dynamodbav:",omitempty"`
+}
+
+var sharedEmptyCollectionsTestCases = []struct {
```

Comment:
When porting this change to v2 we should use named test cases for these to improve investigation of case failures.

Reply:
100% Agree